### PR TITLE
Migrate to different padding for push notification encryption

### PR DESCRIPTION
**Pull request by bbirman**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        let jsonContent = try JSONSerialization.data(withJSONObject: contentDictionary)
+        let encryptedConent = try XCTUnwrap(SFSDKCryptoUtils.aes128EncryptData(jsonContent, withKey: key, iv: iv)).base64EncodedString()
+        
+        // RSA-PKCS1 encryption for secret
+        let secret = key + iv
+        let encryptedSecret = try XCTUnwrap(SFSDKCryptoUtils.encrypt(usingRSAforData: secret, withKeyRef: publicKey))
+        let secretString = encryptedSecret.base64EncodedString()
+        
+        notificationContent.userInfo[kRemoteNotificationKeySecret] = secretString
+        notificationContent.userInfo[kRemoteNotificationKeyContent] = encryptedConent
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift:64]: Typo on encryptedConent

File: libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    
+    func testOAEPSecret() throws {
+        let notificationContent = baseNotificationContent()
+        
+        // Symmetric encryption for payload
+        let key = SFSDKCryptoUtils.randomByteData(withLength: 16)
+        let iv = SFSDKCryptoUtils.randomByteData(withLength: 16)
+        let jsonContent = try JSONSerialization.data(withJSONObject: contentDictionary)
+        let encryptedConent = try XCTUnwrap(SFSDKCryptoUtils.aes128EncryptData(jsonContent, withKey: key, iv: iv)).base64EncodedString()
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift:79]: Typo encryptedConent

File: libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift (hunk 1 above)
Comment (bbirman) [reply libs/SalesforceSDKCore/SalesforceSDKCoreTests/PushNotificationDecryptionTests.swift]: ty!

**Comment by wmathurin** [comment]
Looking at https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m#L565, do you know what the _encodedRSAEncryptionOID is / where it came from?

**Comment by wmathurin** [comment]
It seems like "libraries that use this OID would support both RSAES-PKCS1-v1_5 and RSAES-OAEP for encryption and decryption operations".

**Review by wmathurin — APPROVED** [review]

**Pull request merged**